analyzer/src/main/kotlin/PackageManagerDependencyResult.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2022 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer
+
+data class PackageManagerDependencyResult(
+    /**
+     * The names of package managers that the package manager that provided this result must run before.
+     */
+    val mustRunBefore: Set<String>,
```

Comment:
Why the decision to introduce `mustRunBefore` in addition to `mustRunAfter`? "A mustRunBefore B" can also be modeled as "B mustRunAfter A", but introduces a(nother) way to shoot yourself in the foot by creating contradicting dependencies.

Reply:
It's to give package managers the option to request both, e.g. in the only current use case Pub knows that it must run before Gradle, but Gradle should not have to know about Pub. This is because the requirement is Pub specific.
In the end `mustRunBefore` is translated to `mustRunAfter`, as can be configured by the user.
Note that I plan to introduce cycle detection in a later change to not end up in a deadlock, that's also the reason that the user can overwrite the detected order.

Reply:
I still dislike somewhat that this now doesn't match the `ort.conf` configuration anymore 1:1. Should we then also introduce `mustRunBefore` in `ort.conf` to align?

Reply:
IMO no, because it's possible to configure all possible dependencies with the existing property, I don't see a reason to align the config with the internal API. I initially thought about letting the function just return a `Map<String, String>` which would represent `mustRunAfter`, but I preferred the explicit return type to not have yet another map that is open to interpretation.